ide/coqide/idetop.ml

```diff
@@ -360,6 +360,59 @@ let debug_cmd = ref ""
 let db_cmd cmd =
   debug_cmd := cmd
 
+let db_loc () =
+  let open Loc in
+  let open DebugHook in
+  match debugger_state.cur_loc with
+  | Some {fname=ToplevelInput; bp; ep} ->
+    Some ("ToplevelInput", [bp; ep])
+  | Some {fname=InFile {dirpath=None; file}; bp; ep} ->
+    Some (file, [bp; ep])  (* for Load command *)
+  | Some {fname=InFile {dirpath=Some dirpath}; bp; ep} ->
+    (* todo: check what's in Loc.t on Windows *)
+    let open Names in
+    let dirpath = DirPath.of_string dirpath in
+    let pfx = DirPath.make (List.tl (DirPath.repr dirpath)) in
+    let paths = Loadpath.find_with_logical_path pfx in
+    let basename = match DirPath.repr dirpath with
+    | hd :: tl -> (Id.to_string hd) ^ ".v"
+    | [] -> ""
+    in
+    let vs_files = List.map (fun p -> (Filename.concat (Loadpath.physical p) basename)) paths in
+    let filtered = List.filter (fun p -> Sys.file_exists p) vs_files in
+    begin match filtered with
+    | [] -> (* todo: maybe tweak this later to allow showing a popup dialog in the GUI *)
+      let msg = Pp.(fnl () ++ str "Unable to locate source code for module " ++
+                      str (Names.DirPath.to_string dirpath)) in
+      let msg = if vs_files = [] then msg else
+        (List.fold_left (fun msg f -> msg ++ fnl() ++ str f) (msg ++ str " in:") vs_files) in
+      Feedback.msg_warning msg;
+      None
+    | [f] -> Some (f, [bp; ep])
+    | f :: tl ->
+      let msg = Pp.(fnl () ++ str "Multiple files found matching module " ++
+          str (Names.DirPath.to_string dirpath) ++ str ":") in
+      let msg = List.fold_left (fun msg f -> msg ++ fnl() ++ str f) msg vs_files in
+      Feedback.msg_warning msg;
+      Some (f, [bp; ep]) (* be arbitrary unless we can tell which file was loaded *)
+    end
+  | _ -> None
+
+let db_upd_bpts updates =
+  let open DebugHook in
+  List.iter (fun op ->
+      let ((file, offset), opt) = op in
+      Printf.printf "server:db_upd_bpts '%s': %d %b\n%!" file offset opt;
```

Comment:
Given that this is currently dead code, I have no idea whether this printf is really supposed to be there or if it's a remnant of printf debugging.

Reply:
It was for debugging.  `db_upd_bpts` is used in the next PR.